f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py

```diff
@@ -397,6 +400,14 @@ def __init__(self, conf, registerOpts=True):
                  % (len(self.__bigips), self.conf.icontrol_username))
         LOG.info('iControlDriver dynamic agent configurations:%s'
                  % self.agent_configurations)
+
+        # read enhance services definitions
+        self.esd = EsdTagProcessor('/etc/neutron/services/f5/esd/')
```

Comment:
Added get_config_dir() to determine configuration directory based Oslo config variables: if config_dir define, use that, otherwise get config dir from config file (f5-openstack-agent.ini) path.